### ISSUE TITLE
Question: Errors on Linux but not Mac (same configs)

### ISSUE THREAD

**Issue by atidyshirt** [vimpostor/vim-tpipeline#23]
Hey, this definitely could just be a configuration problem rather than a bug, however I have been using this for quite some time on my Mac computer with no issues, however when transferring my files on Linux, I appear to get the following error sequence:

```conf
Error detected while processing function tpipeline#init_statusline[22]..tpipeline#update[9]..tpipeline#parse#eval_stl:
line    1:
E5555: API call: fillchar must be a single-width character
Press ENTER or type command to continue
Error detected while processing function tpipeline#init_statusline[22]..tpipeline#update[9]..tpipeline#parse#eval_stl:
line    2:
E121: Undefined variable: str
Press ENTER or type command to continue
Error detected while processing function tpipeline#init_statusline[22]..tpipeline#update[9]..tpipeline#parse#eval_stl:
line    4:
E121: Undefined variable: highlights
Press ENTER or type command to continue
Error detected while processing function tpipeline#init_statusline[22]..tpipeline#update[9]..tpipeline#parse#eval_stl:
line    9:
E121: Undefined variable: res
Press ENTER or type command to continue
Error detected while processing function tpipeline#init_statusline[22]..tpipeline#update[9]..tpipeline#parse#eval_stl:
line    9:
E116: Invalid arguments for function substitute
```

Here are my tmux config and tpipeline settings:

```
unbind C-space
unbind C-a
set-option -g prefix C-a
bind C-a send-prefix

# Plugins
set -g @plugin 'tmux-plugins/tmux-resurrect'

# tmux sessionizer
bind-key -r f run-shell "tmux neww -n 'tmux-sessionizer' ~/bin/t"

set -g default-terminal "screen-256color"
set-option -sa terminal-overrides ",tmux-256color"

set-option -g set-titles off
set -g renumber-windows on
set-option -g allow-rename off

set-option -g -q mouse on

# set window panes to start at 1
set -g base-index 1
setw -g pane-base-index 1

setw -g mode-keys vi

bind r source-file ~/.tmux.conf

bind -r H resize-pane -L 5
bind -r J resize-pane -D 5
bind -r K resize-pane -U 5
bind -r L resize-pane -R 5

# Session tree
bind l choose-tree

bind s split-pane -v -c "#{pane_current_path}"
bind v split-pane -h -c "#{pane_current_path}"
bind c new-window -c "#{pane_current_path}"
unbind '"'
unbind %

bind x kill-pane
bind t set status

# vim/tmux key mappings
is_vim="ps -o state= -o comm= -t '#{pane_tty}' \
    | grep -iqE '^[^TXZ ]+ +(\\S+\\/)?g?(view|n?vim?x?)(diff)?$'"
bind-key -n 'C-h' if-shell "$is_vim" 'send-keys C-h'  'select-pane -L'
bind-key -n 'C-j' if-shell "$is_vim" 'send-keys C-j'  'select-pane -D'
bind-key -n 'C-k' if-shell "$is_vim" 'send-keys C-k'  'select-pane -U'
bind-key -n 'C-l' if-shell "$is_vim" 'send-keys C-l'  'select-pane -R'
tmux_version='$(tmux -V | sed -En "s/^tmux ([0-9]+(.[0-9]+)?).*/\1/p")'
if-shell -b '[ "$(echo "$tmux_version < 3.0" | bc)" = 1 ]' \
    "bind-key -n 'C-\\' if-shell \"$is_vim\" 'send-keys C-\\'  'select-pane -l'"
if-shell -b '[ "$(echo "$tmux_version >= 3.0" | bc)" = 1 ]' \
    "bind-key -n 'C-\\' if-shell \"$is_vim\" 'send-keys C-\\\\'  'select-pane -l'"

bind-key -T copy-mode-vi 'C-h' select-pane -L
bind-key -T copy-mode-vi 'C-j' select-pane -D
bind-key -T copy-mode-vi 'C-k' select-pane -U
bind-key -T copy-mode-vi 'C-l' select-pane -R

bind-key -T copy-mode-vi 'C-p' prev
bind-key -T copy-mode-vi 'C-n' next

## styling
set-option -g status "on"
set -g focus-events on
set -g status-justify "centre"
set -g status-left-style "none"
set -g message-command-style "fg=#a89984,bg=#504945"
set -g status-right-style "none"
set -g pane-active-border-style "fg=#a89984"
set -g status-style "none,bg=#3a3735"
set -g message-style "fg=#a89984,bg=#504945"
set -g pane-border-style "fg=#504945"
set -g status-right-length "100"
set -g status-left-length "100"
setw -g window-status-activity-style "underscore,fg=#a89984,bg=#3a3735"
setw -g window-status-separator ""
setw -g window-status-style "none,fg=#a89984,bg=#3a3735"
set -g status-left "#[fg=#282828,bg=#a89984] #S #[fg=#a89984,bg=#504945,nobold,nounderscore,noitalics]#[fg=#a89984,bg=#504945] %R #[fg=#504945,bg=#3a3735,nobold,nounderscore,noitalics]#[fg=#a89984,bg=#3a3735] #{sysstat_mem} #[fg=blue]#{sysstat_ntemp} #[fg=#3a3735,bg=#3a3735,nobold,nounderscore,noitalics]"
set -g status-right '#[fg=#3a3735,bg=#3a3735,nobold,nounderscore,noitalics]#[fg=#a89984,bg=#3a3735] #[fg=blue] #[fg=#504945,bg=#3a3735,nobold,nounderscore,noitalics]#[fg=#a89984,bg=#504945] #(cat #{socket_path}-\#{session_id}-vimbridge) #[fg=#a89984,bg=#504945,nobold,nounderscore,noitalics]#[fg=#282828,bg=#a89984] #H #{prefix_highlight}'
setw -g window-status-format "#[fg=#3a3735,bg=#3a3735,nobold,nounderscore,noitalics]#[fg=#a89984] #I  #W #[fg=#3a3735,bg=#3a3735,nobold,nounderscore,noitalics]"
setw -g window-status-current-format "#[fg=#3a3735,bg=#504945,nobold,nounderscore,noitalics]#[fg=#a89984,bg=#504945] #I  #W #[fg=#504945,bg=#3a3735,nobold,nounderscore,noitalics]"

run-shell ~/.config/tmux/tmux-resurrect/resurrect.tmux
```

```lua
-- tpipeline configuration
vim.g['tpipeline_preservebg'] = 1
vim.g['tpipeline_autoembed'] = 0
vim.g['tpipeline_split'] = 0
vim.g['tpipeline_focuslost'] = 0
vim.cmd [[highlight MyBackground guifg=#a89984 guibg=#504945]]
vim.g['tpipeline_statusline'] = '%#MyBackground#%t'
```

Any help or advice would be appreciated!

**Comment by vimpostor** [comment]
Which version of neovim are you using?

**Comment by atidyshirt** [comment]
on `NVIM v0.7.9.0-dev` on linux and `v0.6.1` on mac

**Comment by vimpostor** [comment]
Neovim regression confirmed, I can reproduce it with `0.7.0` but not with `0.6.1`.
I haven't bisected yet, but if I would have to guess https://github.com/neovim/neovim/pull/17113 might be the problem.

**Comment by atidyshirt** [comment]
Awesome, thanks for the quick reply, for now will just downgrade :)

**Comment by zeertzjq** [comment]
Should be fixed by <https://github.com/neovim/neovim/pull/17666>.

**Comment by vimpostor** [comment]
> Should be fixed by [neovim/neovim#17666](https://github.com/neovim/neovim/pull/17666).

Thanks for fixing this so fast upstream. :)

**Comment by tin-pham** [comment]
Thanks, fixed this by updating lastest neovim 0.7.0dev
